Sources/SwiftLexicalLookup/LookupName.swift

```diff
@@ -135,6 +135,50 @@ import SwiftSyntax
     }
   }
 
+  /// Position of this name.
+  ///
+  /// For some syntax nodes, their position doesn't reflect
+  /// the position at which a particular name was introduced at.
+  /// Such cases are function parameters (as they can
+  /// contain two identifiers) and function declarations (where name
+  /// is precided by access modifiers and `func` keyword).
+  @_spi(Experimental) public var position: AbsolutePosition {
+    switch self {
+    case .identifier(let syntax, _):
+      return syntax.identifier.positionAfterSkippingLeadingTrivia
+    case .declaration(let syntax):
+      return syntax.name.position
+    case .implicit(let implicitName):
+      switch implicitName {
+      case .self(let declSyntax):
+        switch Syntax(declSyntax).as(SyntaxEnum.self) {
+        case .functionDecl(let functionDecl):
+          return functionDecl.name.position
+        case .initializerDecl(let initializerDecl):
+          return initializerDecl.initKeyword.positionAfterSkippingLeadingTrivia
+        case .subscriptDecl(let subscriptDecl):
+          return subscriptDecl.accessorBlock?.position ?? subscriptDecl.endPosition
+        case .variableDecl(let variableDecl):
+          return variableDecl.bindings.first?.accessorBlock?.positionAfterSkippingLeadingTrivia
+            ?? variableDecl.endPosition
+        default:
+          return declSyntax.positionAfterSkippingLeadingTrivia
```

Comment:
Not that it matters much but some cases use `position` and some `positionAfterSkippingLeadingTrivia`. That makes me wonder why there’s a difference. If there is no meaningful distinction, I would suggest that they all use the same, maybe defaulting to `position` because it doesn’t need to add any trivia length.

Reply:
As I tested new lookup with different source code inputs, I’ve discovered that using `positionAfterSkippingLeadingTrivia` over `position` did fix discrepancies in some cases. This is an example of not matching results from `ActiveSyntaxRewriter.swift` after I changed `initializerDeclSyntax` to use `position` for `self` keyword in this `switch`.

```swift
class ActiveSyntaxRewriter: SyntaxRewriter {
  var retainFeatureCheckIfConfigs: Bool // <— line 153

  init(configuredRegions: ConfiguredRegions, retainFeatureCheckIfConfigs: Bool) { // <— line 155
    self.activeClauses = .configuredRegions(configuredRegions) 
    self.diagnostics = activeClauses.priorDiagnostics
    self.retainFeatureCheckIfConfigs = retainFeatureCheckIfConfigs
  }
}
```

```
-----> Lookup started at: 156:5 ("self")
     |      ASTScope      | SwiftLexicalLookup 
> ✅ |configuredRegions 155:8|configuredRegions 155:8
> ✅ |retainFeatureCheckIfConfigs 155:46|retainFeatureCheckIfConfigs 155:46
> ⚠️ |End here: self 155:3|    self 153:40     
> ⏩ |       -----        |  Look memb: 148:7  
```

Do you think there could be some better way to fix this, or should we maybe start using `positionAfterSkippingLeadingTrivia`  consistently?

Reply:
Oh, I see. 

`positionAfterSkippingLeadingTrivia` probably matches the AST’s `position` more closely because the AST does not care about trivia. But I think outside of your validation tooling `position` and `positionAfterSkippingLeadingTrivia` should produce the same lookup results, right? Or can you think of cases where the two produce fundamentally different results?

Anyway, I would suggest to use `positionAfterSkippingLeadingTrivia` consistently for now because it’s also more natural. You really need to think twice about trivia handling to figure out why `self` is declared at `153:40` in the above example, should we use `position`.

Reply:
I agree, there's no fundamental difference between using `position` over `positionAfterSkippingLeadingTrivia`. I changed to `positionAfterSkippingLeadingTrivia` for consistency.